spec/lib/brexit_checker/question_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe BrexitChecker::Question do
+  let(:q1) { FactoryBot.build(:brexit_checker_question) }
+  let(:q2) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(a b) }]) }
+  let(:q3) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(c d) }]) }
+  let(:q4) { FactoryBot.build(:brexit_checker_question, criteria: %w(c)) }
+
+  let(:questions) { [q1, q2, q3, q4] }
+
+  describe '#resolve' do
+    before do
+      allow(BrexitChecker::Question).to receive(:load_all) { questions }
+    end
+
+    let(:subject) {
```

Comment:
Would it be possible to push some of this into the individual 'it' or 'context' blocks? I still find it confusing to see variables like `criteria_keys` set multiple times in the nesting structure, and I'm trying to avoid it myself.

Reply:
removed nested lets.